dandi/dandiapi.py

```diff
@@ -273,11 +276,24 @@ def dandi_authenticate(self):
             raise NotImplementedError("TODO client name derivation for keyring")
         app_id = f"dandi-api-{client_name}"
         keyring_backend, api_key = keyring_lookup(app_id, "key")
-        if not api_key:
-            api_key = input(f"Please provide API Key for {client_name}: ")
-            keyring_backend.set_password(app_id, "key", api_key)
-            lgr.debug("Stored key in keyring")
-        self.authenticate(api_key)
+        if api_key:
+            self.authenticate(api_key)
```

Comment:
not quite... since if there is a known one, we would just try to auth and fail without going through "the loop".

just remove this `if api_key` here, make loop unconditional, ask to provide a key right there only `if not api_key`, set `api_key = None` if auth fails before `continue`.

Reply:
Done.